src/radical/pilot/raptor/worker.py

```diff
@@ -36,61 +43,77 @@ def __init__(self, cfg=None, register=True, session=None):
         self._prof = ru.Profiler(name=self._uid, ns='radical.pilot.worker',
                                  path=self._sbox)
 
-        if register:
+        # register for lifetime management messages on the control pubsub
+        psbox = os.environ['RP_PILOT_SANDBOX']
+        ctrl_cfg = ru.read_json('%s/%s.cfg' % (psbox, rpc.CONTROL_PUBSUB))
 
-            psbox = os.environ['RP_PILOT_SANDBOX']
-            ctrl_cfg = ru.read_json('%s/%s.cfg' % (psbox, rpc.CONTROL_PUBSUB))
+        self._control_sub = ru.zmq.Subscriber(rpc.CONTROL_PUBSUB,
+                                              url=ctrl_cfg['sub'],
+                                              log=self._log,
+                                              prof=self._prof,
+                                              cb=self._control_cb)
 
-            self._control_sub = ru.zmq.Subscriber(rpc.CONTROL_PUBSUB,
-                                                  url=ctrl_cfg['sub'],
-                                                  log=self._log,
-                                                  prof=self._prof,
-                                                  cb=self._control_cb)
+        # we push hertbeat and registration messages on that pubsub also
+        self._ctrl_pub = ru.zmq.Publisher(rpc.CONTROL_PUBSUB,
```

Comment:
just to note - to have same style name - `_control_pub` as for subscriber `_control_sub`

Reply:
Good point - but I removed the `self._control_sub` assignment instead: since that has a callback, the sub handle is never used anyway (and should not be).